openquake/hazardlib/gsim/akkar_bommer_2010.py

```diff
@@ -317,14 +318,17 @@ def get_mean_and_stddevs(self, sites, rup, dists, imt, stddev_types):
         for spec of input and result values.
         """
 
+        ref_sites = copy.deepcopy(sites)
+        ref_sites.vs30 = 600 * np.ones(len(sites.vs30))
```

Comment:
For this GMPE (also Zhao et al and Cauzzi & Faccioli) I am not sure deepcopying is an efficient approach. Given that the SitesContext class is just a simple class holding a numpy array, which you are overwriting, I think it is more efficient to create a new instance of the SitesContext class and add the new values into it e.g.

from openquake.hazardlib.gsim.base import SitesContext
...
ref_sites = SitesContext()
ref_site.vs30 = 600.0 * np.ones_like(sites.vs30)

I did a quick profiling of the two approaches and here are my results
```python
import numpy as np
from copy import deepcopy
from openquake.hazardlib.gsim.base import SitesContext
trial_vs30 = np.random.uniform(100., 1000., 50000)
ctx1 = SitesContext()
ctx1.vs30 = trial_vs30
ref_vs30 = 600.0 * np.ones_like(ctx1.vs30)
def modify_context1(ctx1, new_vs30):
    ctx2 = deepcopy(ctx1)
    ctx2.vs30 = new_vs30
    return ctx2

def modify_context2(ctx1, new_vs30):
    ctx2 = SitesContext()
    ctx2.vs30 = new_vs30
    return ctx2

timeit ctx2 = modify_context1(ctx1, ref_vs30)
10000 loops, best of 3: 68.3 µs per loop

timeit ctx2 = modify_context2(ctx1, ref_vs30)
1000000 loops, best of 3: 368 ns per loop
```